karma.conf.js

```diff
@@ -0,0 +1,107 @@
+'use strict';
+
+var depMain = function(name) {
+  var pkg = require(name + '/package.json');
+  var main = 'main' in pkg ? pkg.main : 'index.js';
+  return './node_modules/' + name + '/' + main;
+};
+
+// https://saucelabs.com/platforms
+var customLaunchers = {
+  sl_chrome: {
+    base: 'SauceLabs',
+    browserName: 'chrome',
+    version: '51'
+  },
+  sl_firefox: {
+    base: 'SauceLabs',
+    browserName: 'firefox',
+    version: '47'
+  },
+  sl_Safari_8: {
+    base: 'SauceLabs',
+    browserName: 'safari',
+    platform: 'OS X 10.10',
+    version: '8'
+  },
+  SL_Safari_10: {
+    base: 'SauceLabs',
+    browserName: 'safari',
+    platform: 'OS X 10.11',
+    version: '10'
+  },
+  SL_iOS_Safari_8: {
+    base: 'SauceLabs',
+    browserName: 'iphone',
+    version: '8.1'
+  },
+  SL_IE_9: {
+    base: 'SauceLabs',
+    browserName: 'internet explorer',
+    platform: 'Windows 2008',
+    version: '9'
+  },
+  sl_ie_11: {
+    base: 'SauceLabs',
+    browserName: 'internet explorer',
+    platform: 'Windows 8.1',
+    version: '11'
+  }
+};
+
+var options = {
+
+  browserDisconnectTimeout: 10000,
+  browserDisconnectTolerance: 2,
+  browserNoActivityTimeout: 90000,
+  captureTimeout: 120000,
+
+  client: {
+    mocha: {opts: 'test/mocha.opts'}
+  },
+
+  frameworks: [
+    'browserify',
+    'mocha'
+  ],
+
+  files: Object.keys(require('./package.json').dependencies)
+    .map(depMain)
+    .concat(['index.js', 'test/**/*.js']),
```

Comment:
We now referencing `require('./package.json').dependencies` in two places. We could define `dependencies` near the top of the file:

```javascript
//  dependencies :: Array String
var dependencies = Object.keys(require('./package.json').dependencies);
```

This would allow the following changes:

```diff
-  files: Object.keys(require('./package.json').dependencies)
-    .map(depMain)
-    .concat(['index.js', 'test/**/*.js']),
+  files: dependencies.map(depMain).concat(['index.js', 'test/**/*.js']),
```

```diff
-        for (var name in require('./package.json').dependencies) {
-          bundle.require(depMain(name), {expose: name});
-        }
+        dependencies.forEach(function(name) {
+          bundle.require(depMain(name), {expose: name});
+        });
```